### ISSUE TITLE
Unhandled rejection 

### ISSUE THREAD

**Issue by gregory-h** [Novaleaf/phantomjscloud-node#7]
Using the following client code if options.url points to a webserver that is down an unhandled rejection is thrown and the process dies. Client can't catch or otherwise handle the error.  Error originates in the http lib axios at /lib/core/createError.js:15 

```
var browser = new phantomJsCloud.BrowserApi(config.KEY);

var options = {
    url: url,
    renderType: 'html'
}   

browser
.requestSingle(options)
.then(function(response){
    if( response.statusCode === 200 ){
        cb(null, response.content.data);
    } else {
        cb( response.statusCode )
    }
},function(err) {
    cb(err)
}).catch(function(err) {
    cb(err)            
}) ```

**Comment by jasonswearingen** [comment]
thank you, this looks to be because the http library we are using isn't fully compatabile with our Promises library.  will investigate and hopefully deploy a fix today.

**Comment by jasonswearingen** [comment]
this has been fixed in 2.0.4 (published) and tests added.
